### DISCUSSION TITLE
How do I change the link color of a node for which I have id information at a certain moment?

### DISCUSSION THREAD

**Discussion by furkanbicak** [xyflow/xyflow#3509] [Q&A]
### Please describe the feature that you want to propose

My connection type between nodes is gray by default, and I want to instantly change the connection color of a node for which I have id information to a different color. How can I do this?

**Comment by moklick** [comment]
Do you want to colorize the handles or the connection line? Both things are possible. You can use `isValidConnection` in combination with the "valid" and "connecting" class names as seen in this example https://reactflow.dev/docs/examples/interaction/validation/ or use `useStore` in a custom node to access an ongoing connection:

```js 
const { nodeId, startHandle } = useStore(s => ({ startHandle: s.connectionStartHandle, nodeId: s.connectionNodeId }))
```

In v12 we will add a `useConnection` hook to simplify stuff like this.
